pkg/challenges/challenges.go

```diff
@@ -58,25 +59,29 @@ func PrincipalFromIDToken(ctx context.Context, tok *oidc.IDToken) (identity.Prin
 	}
 	var principal identity.Principal
 	var err error
-	switch iss.Type {
-	case config.IssuerTypeBuildkiteJob:
-		principal, err = buildkite.JobPrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeGitLabPipeline:
-		principal, err = gitlabcom.JobPrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeEmail:
-		principal, err = email.PrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeSpiffe:
-		principal, err = spiffe.PrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeGithubWorkflow:
-		principal, err = github.WorkflowPrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeKubernetes:
-		principal, err = kubernetes.PrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeURI:
-		principal, err = uri.PrincipalFromIDToken(ctx, tok)
-	case config.IssuerTypeUsername:
-		principal, err = username.PrincipalFromIDToken(ctx, tok)
-	default:
-		return nil, fmt.Errorf("unsupported issuer: %s", iss.Type)
+	if iss.IsCiProvider {
```

Comment:
Instead of adding a new flag, can we use `iss.Type`? The `Type` of each CI OIDC provider could be `"CIProvider"`. Then the new issuer would simply be another case statement.

Reply:
I agree! I was just wondering how could we know the real type of the provider after migrating all of them, because we should need know what claim mapping use for each provider. But I think it will work by using the provider name as string and not as a type.